cql3/type_json.cc

```diff
@@ -75,6 +77,31 @@ static sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+template <std::floating_point Tf, std::integral Ti>
+static std::partial_ordering cmp(Tf x, Ti y) {
+    return x <=> y;
+}
+
+// we are not able to fit an uint64_t or a 64 bit float into a wider numeric type
+// to compare them, so need to do this in a more convoluted way. specifically, we
+// have to compare the float with the maximum value represented by the integer
+// type to see if the nearest conversion changes the result of the comparison.
+//
+// see https://github.com/JuliaLang/julia/blob/49b361aa2ec68efe4cb0d5870e772eb3e0d89993/base/float.jl#L553-L565
+template <std::floating_point Tf, std::integral Ti>
+requires (sizeof(Tf) <= 8 && sizeof(Ti) >= 8)
+static std::partial_ordering cmp(Tf x, Ti y) {
+    auto fy = static_cast<Tf>(y);
+    if (x < fy || (x == fy && (fy < static_cast<Tf>(std::numeric_limits<Ti>::max()) &&
+                               std::bit_cast<Ti>(fy) < y))) {
```

Comment:
I don't understand the role of the various pieces of this logic (and in particular what this bit_cast does). A comment explaining or giving examples would have helped, but perhaps not necessary if you have good tests covering these exact code paths. I'll write more below about the tests.